alf/algorithms/mcts_algorithm.py

```diff
@@ -338,9 +379,35 @@ def predict_step(self, time_step: TimeStep, state: MCTSState):
         # will be based on none-zero ucb_scores
         trees.visit_count[roots] = 1
         trees.value_sum[roots] = model_output.value
-        self._update_best_child(trees, roots)
+        if not self._expand_all_children:
+            self._build_tree1(trees, to_plays)
+        else:
+            self._build_tree2(trees, to_plays)
 
-        for sim in range(1, self._num_simulations + 1):
+        action, info = self._select_action(trees, steps)
+        return AlgStep(
+            output=action, state=MCTSState(steps=state.steps + 1), info=info)
+
+    def _build_tree1(self, trees, to_plays):
+        """Build the tree by evaluating one node a time"""
+        roots = (trees.B, trees.root_indices)
+        sim0 = 1
+        if self._expand_all_root_children:
+            v = self._expand_children(trees, roots, 1, to_plays)
+            branch_factor = trees.branch_factor
+            trees.visit_count[roots] += branch_factor
```

Comment:
Suggest adding a comments here saying that we are effectively backing up the expansion result (of the roots' children)?